test/conformance/serial/ptp.go

```diff
@@ -884,3 +1140,69 @@ func testCaseEnabled(testCase TestCase) bool {
 	}
 	return false
 }
+
+func processRunning(input string) (map[string]bool, error) {
+	// Regular expression pattern
+	processStatusPattern := `openshift_ptp_process_status\{config="([^"]+)",node="([^"]+)",process="([^"]+)"\} (\d+)`
+
+	// Compile the regular expression
+	processStatusRe := regexp.MustCompile(processStatusPattern)
+
+	// Find matches
+	processRunning := map[string]bool{"phc2sys": false, "ptp4l": false, "ts2phc": false, "gpspipe": false, "gpsd": false}
+
+	scanner := bufio.NewScanner(strings.NewReader(input))
+	timeout := 1 * time.Minute
+	start := time.Now()
+	for scanner.Scan() {
+		t := time.Now()
+		elapsed := t.Sub(start)
+		if elapsed > timeout {
+			return map[string]bool{}, fmt.Errorf("timedout scanning metrics for clockState values")
+		}
+		line := scanner.Text()
+		if matches := processStatusRe.FindStringSubmatch(line); matches != nil {
+			if _, ok := processRunning[matches[3]]; ok && matches[4] == "1" {
+				processRunning[matches[3]] = true
+			}
+
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		fmt.Println("Error reading input:", err)
+	}
+	return processRunning, nil
+}
+
+func clockStateByProcesses(input string) (map[string]bool, error) {
+	// Regular expression pattern
+	clockStatePattern := `openshift_ptp_clock_state\{iface="([^"]+)",node="([^"]+)",process="([^"]+)"\} (\d+)`
+
+	// Compile the regular expression
+	processStatusRe := regexp.MustCompile(clockStatePattern)
+
+	// Find matches
+	processClockState := map[string]bool{"phc2sys": false, "GM": false, "dpll": false, "ts2phc": false, "gnss": false}
+
+	scanner := bufio.NewScanner(strings.NewReader(input))
+	timeout := 1 * time.Minute
+	start := time.Now()
+	for scanner.Scan() {
+		t := time.Now()
+		elapsed := t.Sub(start)
+		if elapsed > timeout {
+			return map[string]bool{}, fmt.Errorf("timedout scanning metrics for clockState values")
```

Comment:
break here instead. might want to reduce the timeout to 10 seconds.

Reply:
not sure why break rather return is better I changed it to return nil, error.. the test has an assertion for the error to be nil so the test would fail if this is hit.